src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,137 @@
+"""Large language model (LLM) based policy guided bilevel planning approach.
+
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+import time
+from typing import Dict, FrozenSet, List, Optional, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.settings import CFG
+from predicators.src.structs import GroundAtom, Object, ParameterizedOption, \
+    Sequence, State, Task, _GroundNSRT
+
+
+class LLMProbeApproach(OpenLoopLLMApproach):
+    """LLMProbeApproach definition."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "llm_probe"
+
+    def _get_llm_based_plan(
+            self, state: State, atoms: Set[GroundAtom],
+            goal: Set[GroundAtom]) -> Optional[List[_GroundNSRT]]:
+        """Query the LLM and use the outputs to initialize the queue in A*."""
+        # To initialize the queue in A*, we will build up a partial policy
+        # and then run policy-guided A*.
+        start_time = time.time()
+        objects = set(state)
+        partial_policy_dict: Dict[FrozenSet[GroundAtom], _GroundNSRT] = {}
+        for option_plan in self._get_llm_based_option_plans(
```

Comment:
i think that all 3 settings in settings.py that start with "open_loop_llm" should be changed to just start with "llm", since all 3 are used in this new approach too, which is not open-loop